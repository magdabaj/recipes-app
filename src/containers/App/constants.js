export const LOAD_RECIPES = 'app/App/LOAD_RECIPES';
export const LOAD_RECIPES_SUCCESS = 'app/App/LOAD_RECIPES_SUCCESS';
export const LOAD_RECIPES_ERROR = 'app/App/LOAD_RECIPES_ERROR';

export const LOAD_TAGS = 'app/App/LOAD_TAGS';
export const LOAD_TAGS_SUCCESS = 'app/App/LOAD_TAGS_SUCCESS';
export const LOAD_TAGS_ERROR = 'app/App/LOAD_TAGS_ERROR';