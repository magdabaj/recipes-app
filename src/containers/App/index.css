body {
    box-sizing: border-box;
    margin: 0 auto;
    /*margin-left: 0;*/

    text-align: center;
    width: 100%;
    max-width: 1170px;

    font-size: 16px;
    font-family: Open Sans, sans-serif;
    font-weight: 400;
    line-height: 30px;
    color: #303030;
    display: flex;
    flex-direction: column;
}

#app {
}

h1 {
    font-size: 4vw;
}

h2 {
    font-size: 2vw;
}

h3 {
    font-size: 1.5vw;
}

img {
    width: 100%;
    height: auto;
    min-height: 300px;
    /*max-height: 400px;*/
}

a {
    text-decoration: none;
    color: black;
}

.footer {
    color: white;
    background-color: #ee4c7c;
    padding: 65px 0 60px;
    background-size: cover;
    width: 100%;
    margin-top: 20px;
    min-height: 500px;
    flex-shrink: 0;
    bottom: 0;
    margin: auto;
}

.footer-container {
    /*opacity: 0.5;*/
}

.input {
    border-bottom-color: rgba(0, 0, 0, 0.2);
    border-bottom-width: 1px;
    border-bottom-style: solid;
    padding: 5px;
    border-top: 0;
    border-left: 0;
    border-right: 0;
    width: 100%;
    font-size: 18px;
}

.input:focus {
    outline: none;
    border-bottom-color: #ee4c7c;
    border-bottom-width: 2px;
}


.input-error {
    border-bottom-color: #ee4c7c;
}

.error-message {
    color: #ee4c7c;
    line-height: 16px;
    margin-top: 5px;
    font-size: 14px;
}

.label {
    min-width: 150px;
    font-size: 16px;
    font-weight: 700;
}

.form-empty-div-1 {
    width: 15%;
}
.form-empty-div-2 {
    width: 15%;
}

.form-container {
    max-width: 1170px;
    padding: 50px;
    display: flex;
    justify-content: center;
    align-items: center;
}

.form {
    background-color: #fff;
    box-shadow: rgba(0, 0, 0, 0.12) 0px 1px 6px, rgba(0, 0, 0, 0.12) 0px 1px 4px;
    border-radius: 2px;
    padding: 20px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.sign-up {
    display: flex;
    /*justify-content: center;*/
    align-items: center;
}

.login-label {
    text-align: center;
    padding: 0;
    font-size: 13px;
    font-weight: 600;
}

.input-data {
    margin: 5px 0;
}

@media only screen and (min-width: 768px) {
    /*.form {*/
    /*    width: 100%;*/
    /*    padding: 20px;*/
    /*}*/

    /*.input-data {*/
    /*    width: 100%;*/
    /*}*/


    .sign-up {
        justify-content: center;
    }


    /*.form-container {*/
    /*    margin-top: 50%;*/
    /*}*/
}

@media only screen and (max-width: 768px) {
    .form {
        width: 100%
    }

    .input-data {
        width: 100%;
    }

    .sign-up {
        justify-content: space-between;
        /*width: 100%;*/
    }

    .footer-container {
        width: 50%
    }


}
