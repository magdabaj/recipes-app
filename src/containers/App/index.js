import React from 'react';
import { Switch, Route } from 'react-router-dom';
import 'react-toastify/dist/ReactToastify.css';
import HomePage from "../HomePage/Loadable";
import NavigationContainer from "../NavigationContainer";
import LoginContainer from "../LoginContainer";
import UserRecipesContainer from "../UserRecipesContainer/Loadable";
import RecipesFormContainer from "../RecipesFormContainer/Loadable";
import './index.css'
import RecipeContainer from "../RecipeContainer/Loadable";
import {ToastContainer} from "react-toastify";
// todo add button go to all recipes

export default function App() {
    return (
        <div>
            <NavigationContainer />
            <Switch>
                <Route exact path="/" component={HomePage} />
                <Route path={'/home/tag/:tagId/page/:page'} component={HomePage} />
                <Route path={'/home/page/:page'} component={HomePage} />
                <Route path={'/home/tag/:tagId'} component={HomePage} />
                <Route path="/home" component={HomePage} />
                <Route path="/recipes/:recipeId" component={RecipeContainer} />
                <Route path={'/add/:userId/recipe/:recipeId'} component={RecipesFormContainer} />
                <Route path={'/add/:userId'} component={RecipesFormContainer} />
                <Route path={'/user-recipes/page/:page'} component={UserRecipesContainer} />
                <Route path={'/user-recipes'} component={UserRecipesContainer} />
                <Route path="/login" component={LoginContainer} />
                {/*<Route component={NotFoundPage} />*/}
            </Switch>
            <ToastContainer />
            {/*<FooterContainer />*/}
            {/*<GlobalStyle />*/}
        </div>
    );
}
