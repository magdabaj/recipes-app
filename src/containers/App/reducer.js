/*
 *
 * RecipesHomePage reducer
 *
 */
import produce from 'immer';
import fetchStates from '../../utils/fetchStates';
import {
    LOAD_TAGS,
    LOAD_TAGS_ERROR,
    LOAD_TAGS_SUCCESS,
} from './constants';

export const initialState = {
    recipes: [],
    tags: [],
    status: fetchStates.fetching,
    error: null,
    page: 1,
    totalPages: null,
    previousPage: null,
    nextPage: null,
};

/* eslint-disable default-case, no-param-reassign */
const appReducer = (state = initialState, action) =>
    produce(state, draft => {
        switch (action.type) {
            case LOAD_TAGS:
                draft.status = fetchStates.fetching;
                break;
            case LOAD_TAGS_ERROR:
                draft.status = fetchStates.error;
                draft.error = action.error;
                break;
            case LOAD_TAGS_SUCCESS:
                draft.status = fetchStates.success;
                draft.tags = action.tags;
                break;
        }
    });

export default appReducer;
