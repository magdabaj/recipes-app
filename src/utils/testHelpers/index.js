import { Router } from 'react-router-dom'
import { createMemoryHistory } from "history";
import store from "../store";
import {Provider} from "react-redux";
import React from "react";
import { render } from "@testing-library/react";
import {createStore} from "redux";
// import {reducer} from "react-toastify/dist/hooks/toastContainerReducer";

// const store  = createStore(rootReducer, initialState={})
const memoryHistory = createMemoryHistory({
    initialEntries: ['/'],
})

export const withStoreAndRouter = component =>
    <Router history={memoryHistory}>
        <Provider store={store}>
            {component}
        </Provider>
    </Router>

export const renderWithRouter = (
    ui,
    {
        history = memoryHistory,
    } = {}
) => (
    {
        ...render(<Router history={history}>{ui}</Router>),
        history,
    }
)

export const renderWithRouterKentCDodds = (
    ui,
        {
            route='/',
            history = createMemoryHistory({initialEntries: [route]}),
            ...renderOptions
        } = {}
) => {
    // eslint-disable-next-line react/prop-types
    function Wrapper({children}) {
        return <Router history={history}>
            {children}
        </Router>
    }

    return {
        ...render(ui, {wrapper: Wrapper, ...renderOptions})
    }
}

export const renderWithProviderKentCDodds = (
        ui,
        reducer,
        {initialState, store = createStore(reducer, initialState), ...options} = {},
    ) => {
    // eslint-disable-next-line react/prop-types
    function Wrapper({children}) {
        return <Provider store={store}>
            {children}
        </Provider>
    }
    return render(ui, {wrapper: Wrapper, ...options})
}