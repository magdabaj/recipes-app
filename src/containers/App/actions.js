import {LOAD_RECIPES_ERROR, LOAD_TAGS, LOAD_TAGS_ERROR, LOAD_TAGS_SUCCESS} from "./constants";

// export const loadRecipes = page => ({
//     type: LOAD_RECIPES,
//     page,
// });
//
// export const loadRecipesSuccess = response => ({
//     type: LOAD_RECIPES_SUCCESS,
//     response,
// });

export const loadRecipesError = error => ({
    type: LOAD_RECIPES_ERROR,
    error,
});

export const loadTags = () => ({
    type: LOAD_TAGS,
});

export const loadTagsSuccess = tags => ({
    type: LOAD_TAGS_SUCCESS,
    tags,
});

export const loadTagsError = error => ({
    type: LOAD_TAGS_ERROR,
    error,
});