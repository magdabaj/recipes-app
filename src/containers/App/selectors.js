import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the recipesHomePage state domain
 */

const selectAppDomain = state => state.appReducer || initialState;

// const selectAppRoute = (state, props) => props.match.params.tagId;
//
// const selectTagId = () =>
//     createSelector(
//         selectAppRoute,
//         tagId => tagId,
//     );
//
// const selectHomePagePaging = (state, props) => props.match.params.page;
//
// const selectPage = () =>
//     createSelector(
//         makeSelectRecipesHomePage(),
//         selectHomePagePaging,
//         (homeState, pageRoute) => (pageRoute ? pageRoute : homeState.page),
//     );

/**
 * Other specific selectors
 */

/**
 * Default selector used by RecipesHomePage
 */

const makeSelectApp = () =>
    createSelector(
        selectAppDomain,
        substate => substate,
    );

// const makeSelectRecipes = () =>
//     createSelector(
//         selectRecipesHomePageDomain,
//         substate => substate.recipes,
//     );

const makeSelectTags = () =>
    createSelector(
        selectAppDomain,
        substate => substate.tags,
    );

// const makeSelectTotalPages = () =>
//     createSelector(
//         selectRecipesHomePageDomain,
//         substate => substate.totalPages,
//     );
//
// const makeSelectNextPage = () =>
//     createSelector(
//         selectRecipesHomePageDomain,
//         substate => substate.nextPage,
//     );
//
// const makeSelectPreviousPage = () =>
//     createSelector(
//         selectRecipesHomePageDomain,
//         substate => substate.previousPage,
//     );
//
// const makeSelectStatus = () =>
//     createSelector(
//         selectRecipesHomePageDomain,
//         substate => substate.status,
//     );

export default makeSelectApp;
export {
    makeSelectTags
};
