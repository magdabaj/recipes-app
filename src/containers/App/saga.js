import {call, put, takeLatest} from "redux-saga/effects";
import {fetchAllTags} from "../../utils/api/tags";
import {loadTagsError, loadTagsSuccess} from "./actions";
import {LOAD_TAGS} from "./constants";

export function* loadTagsSaga() {
    try {
        const tags = yield call(fetchAllTags);
        yield put(loadTagsSuccess(tags));
    } catch (error) {
        yield put(loadTagsError(error.message));
    }
}

export function* watchTagsLoad() {
    yield takeLatest(LOAD_TAGS, loadTagsSaga);
}