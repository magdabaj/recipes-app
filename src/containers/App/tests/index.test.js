import React from 'react';

import App from '../index';
import { withStoreAndRouter} from "../../../utils/testHelpers";
import {describe, expect, it} from "@jest/globals";
import { render } from "@testing-library/react";


function renderApp(args) {
    const defaultProps = {
        authenticate: () => {},
        user: {},
    }
    const props = {
        ...defaultProps,
        ...args
    }

    return render(withStoreAndRouter(<App {...props} />))
}

describe('<App/>', () => {
    it('should not log errors in console', function () {
        const spy = jest.spyOn(global.console, 'error')
        renderApp()
        expect(spy).not.toHaveBeenCalled()
    });

    it('should render and match the snapshot', () => {
        const { container } = renderApp()
        expect(container.firstChild).toMatchSnapshot()
    })
})