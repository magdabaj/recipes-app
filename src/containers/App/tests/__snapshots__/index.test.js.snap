// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should render and match the snapshot 1`] = `
<div>
  <div
    class="navigation-container"
  >
    <div
      class="navigation"
    >
      <div
        class="navigation-title"
      >
        <a
          aria-current="page"
          class="navigation-title--item active"
          href="/"
        >
          Przepisy
        </a>
      </div>
      <div
        class="navigation-links"
      >
        <a
          aria-current="page"
          class="navigation-links--item navigation-links--active"
          href="/"
        >
          Strona główna
        </a>
      </div>
      <div
        class="navigation-login"
      >
        <a
          class="navigation-login--item"
          href="/login"
        >
          Zaloguj
        </a>
      </div>
    </div>
  </div>
  <div
    class="Spinner__Wrapper-i95by0-0 kvrnfL"
  >
    <div
      style="fill: #9A1750; height: 50%; width: 50%;"
    >
      <svg
        viewBox="0 0 32 32"
        xmlns="http://www.w3.org/2000/svg"
      >
        
  
        <circle
          cx="16"
          cy="3"
          r="0"
        >
          
    
          <animate
            attributeName="r"
            begin="0"
            calcMode="spline"
            dur="1s"
            keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
            repeatCount="indefinite"
            values="0;3;0;0"
          />
          
  
        </circle>
        
  
        <circle
          cx="16"
          cy="3"
          r="0"
          transform="rotate(45 16 16)"
        >
          
    
          <animate
            attributeName="r"
            begin="0.125s"
            calcMode="spline"
            dur="1s"
            keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
            repeatCount="indefinite"
            values="0;3;0;0"
          />
          
  
        </circle>
        
  
        <circle
          cx="16"
          cy="3"
          r="0"
          transform="rotate(90 16 16)"
        >
          
    
          <animate
            attributeName="r"
            begin="0.25s"
            calcMode="spline"
            dur="1s"
            keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
            repeatCount="indefinite"
            values="0;3;0;0"
          />
          
  
        </circle>
        
  
        <circle
          cx="16"
          cy="3"
          r="0"
          transform="rotate(135 16 16)"
        >
          
    
          <animate
            attributeName="r"
            begin="0.375s"
            calcMode="spline"
            dur="1s"
            keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
            repeatCount="indefinite"
            values="0;3;0;0"
          />
          
  
        </circle>
        
  
        <circle
          cx="16"
          cy="3"
          r="0"
          transform="rotate(180 16 16)"
        >
          
    
          <animate
            attributeName="r"
            begin="0.5s"
            calcMode="spline"
            dur="1s"
            keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
            repeatCount="indefinite"
            values="0;3;0;0"
          />
          
  
        </circle>
        
  
        <circle
          cx="16"
          cy="3"
          r="0"
          transform="rotate(225 16 16)"
        >
          
    
          <animate
            attributeName="r"
            begin="0.625s"
            calcMode="spline"
            dur="1s"
            keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
            repeatCount="indefinite"
            values="0;3;0;0"
          />
          
  
        </circle>
        
  
        <circle
          cx="16"
          cy="3"
          r="0"
          transform="rotate(270 16 16)"
        >
          
    
          <animate
            attributeName="r"
            begin="0.75s"
            calcMode="spline"
            dur="1s"
            keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
            repeatCount="indefinite"
            values="0;3;0;0"
          />
          
  
        </circle>
        
  
        <circle
          cx="16"
          cy="3"
          r="0"
          transform="rotate(315 16 16)"
        >
          
    
          <animate
            attributeName="r"
            begin="0.875s"
            calcMode="spline"
            dur="1s"
            keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
            repeatCount="indefinite"
            values="0;3;0;0"
          />
          
  
        </circle>
        
  
        <circle
          cx="16"
          cy="3"
          r="0"
          transform="rotate(180 16 16)"
        >
          
    
          <animate
            attributeName="r"
            begin="0.5s"
            calcMode="spline"
            dur="1s"
            keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
            repeatCount="indefinite"
            values="0;3;0;0"
          />
          
  
        </circle>
        

      </svg>
      

    </div>
  </div>
  <div
    class="Toastify"
  />
</div>
`;
